src/service_daemon.rs

```diff
@@ -2223,7 +2446,24 @@ fn check_service_name(fullname: &str) -> Result<()> {
     Ok(())
 }
 
-fn call_listener(
+/// Validate a hostname.
+fn check_hostname(hostname: &str) -> Result<()> {
+    if !hostname.ends_with(".local.") {
+        return Err(e_fmt!("Hostname must end with '.local.'"));
+    }
+
+    if hostname.is_empty() {
```

Comment:
This check is not needed because the previous check already made sure it's not empty

Reply:
Right, I'll check that there's something before `.local.` instead.

On a side note, should I run `check_hostname` on the hostname given to `ServiceInfo`? I had some trouble earlier while testing this, where I had given `mymachine.local` (without the last dot) to the service announcer, and then trying to resolve the hostname `mymachine.local.`

Reply:
I think it's good to run the same check.  It might triggers a few test case failures though as we didn't enforce it so far. If you have time to make the changes, that will be great! Otherwise, it's perfectly fine if you open a new issue to track it.

Reply:
I have added the hostname check to registered services, and fixed the tests. But note that it might be a breaking change for some users.

Reply:
Thanks for doing this extra! You're right, this will be a breaking change. I will document this in the next version release. It's a trade-off but I think overall it's good to change it. (If any surprise happens, we will handle it ;-))